### PR TITLE
[wgsl] intrinsic and derivative expression

### LINKED ISSUE #603: [wgsl] intrinsics and their arguments
In the current grammar, intrinsic functions like `IsNan` are specified within the `unary_expression` rule, which seems confusing to me. Another problem is their arguments being `IDENT` instead of some more general kind of expression.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #603 
Names are subject to discussion:)

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -1134,15 +1134,44 @@ postfix_expression
...
+  : unary_expression
+  | intrinsic_expression
+  | derivative_function (LESS_THAN derivative_modifier GREATER_THAN)? PAREN_LEFT IDENT PAREN_RIGHT
+
 unary_expression
   : primary_expression postfix_expression
   | MINUS unary_expression
       OpSNegate
       OpFNegate
   | BANG unary_expression
```
Comment (dj2) [inline wgsl/index.bs]: This means you can't do `!IsAny()` anymore.

Comment (kvark) [reply wgsl/index.bs]: yeah, I'll need to refactor this bit

Comment (kvark) [reply wgsl/index.bs]: @dj2 how about now?

**Comment by kvark** [comment]
@litherum as a possible future spec editor for WGSL, do you also want to have a look? Or should we just land changes more freely for now?

**Comment by jdashg** [comment]
I'm surprised to see these in the grammar, and not just in a big set of builtin functions.

**Comment by dj2** [comment]
I'm going to land this. It isn't adding or removing anything, just clarifying the grammar.

**Comment by dj2** [comment]
@jdashg if there is a better way to do it, please open a bug. They seemed to make sense to me as they were directly translating to SPIR-V, but if we should do it in another way I'm open to suggestions.

**Comment by jdashg** [comment]
> 
> 
> @jdashg if there is a better way to do it, please open a bug. They seemed to make sense to me as they were directly translating to SPIR-V, but if we should do it in another way I'm open to suggestions.

#616

**Pull request merged**